app/repositories/UpscanMongoRepository.scala

```diff
@@ -16,83 +16,89 @@
+          .name("reference")
+          .unique(true)
+      ),
+      model.IndexModel(
+        keys = ascending("reference", "registrationId"),
+        indexOptions = IndexOptions()
+          .name("referenceAndRegistrationId")
+          .unique(true)
+      ),
+      model.IndexModel(
+        keys = ascending("timestamp"),
+        indexOptions = IndexOptions()
+          .name("TTL")
+          .expireAfter(backendConfig.expiryInSeconds, TimeUnit.SECONDS)
+      )
+    )
   ) {
 
   val referenceKey = "reference"
   val registrationIdKey = "registrationId"
 
-  override def indexes: Seq[Index] = Seq(
-    Index(
-      name = Some("reference"),
-      key = Seq("reference" -> IndexType.Ascending),
-      unique = true
-    ),
-    Index(
-      name = Some("referenceAndRegistrationId"),
-      key = Seq(
-        "reference" -> IndexType.Ascending,
-        "registrationId" -> IndexType.Ascending
-      ),
-      unique = true
-    ),
-    Index(
-      name = Some("TTL"),
-      key = Seq(
-        "timestamp" -> IndexType.Ascending
-      ),
-      unique = false,
-      options = BSONDocument("expireAfterSeconds" -> BSONInteger(backendConfig.expiryInSeconds))
-    )
-  )
-
-  def getUpscanDetails(reference: String): Future[Option[UpscanDetails]] =
-    find(referenceKey -> JsString(reference))
-      .map(_.headOption)
+  def getUpscanDetails(reference: String): Future[Option[UpscanDetails]] = {
+    collection
+      .find(filter = equal(referenceKey, reference))
+      .toFuture().map(_.headOption)
+  }
 
   def getAllUpscanDetails(registrationId: String): Future[Seq[UpscanDetails]] =
-    find(registrationIdKey -> JsString(registrationId))
+    collection
+      .find(filter = equal(registrationIdKey, registrationId))
+      .toFuture()
 
-  def deleteUpscanDetails(reference: String): Future[Boolean] =
-    remove(referenceKey -> JsString(reference)).map(_.ok)
+  def deleteUpscanDetails(reference: String): Future[Boolean] = {
+    collection
+      .deleteOne(filter = equal(referenceKey, reference))
+      .toFuture().map(_.wasAcknowledged())
+  }
 
   def deleteAllUpscanDetails(registrationId: String): Future[Boolean] =
-    remove(registrationIdKey -> JsString(registrationId)).map(_.ok)
+    collection
+      .deleteMany(filter = equal(registrationIdKey, registrationId))
+      .toFuture().map(_.wasAcknowledged())
 
   def upsertUpscanDetails(upscanDetails: UpscanDetails): Future[UpscanDetails] = {
-    val selector = Json.obj(referenceKey -> upscanDetails.reference)
-    val modifier = Json.obj("$set" -> (
-      Json.toJson(upscanDetails).as[JsObject] ++
-        Json.obj(
-          "timestamp" -> Json.obj("$date" -> timeMachine.timestamp.toInstant(ZoneOffset.UTC).toEpochMilli)
-        )
-      ))
+    val upscanList = Seq(
+      Updates.set("timestamp", timeMachine.timestamp)
+    ) ++ Json.toJson(upscanDetails).as[JsObject]
+      .fields.map { case (key, value) => Updates.set(key, Codecs.toBson(value)) }
```

Comment:
Hi Kaizar, I thought this didn't work when you were going through manual tests (even though the unit and integration tests were fine)?